src/pages/Admin/Admin.tsx

```diff
@@ -78,9 +76,11 @@ const VenueList: React.FC<VenueListProps> = ({
   roomIndex,
 }) => {
   // @debt This selector relies on all venues in firebase being loaded into memory.. not very efficient
-  const venues = useSelector(orderedVenuesSelector);
+  const { isRelatedVenuesLoading, relatedVenues } = useRelatedVenues({
+    currentVenueId: selectedVenueId,
+  });
 
-  if (!venues) return <>Loading...</>;
+  if (!isRelatedVenuesLoading) return <>Loading...</>;
```

Comment:
[nit] Replace this with `Loading` or `LoadingPage` components as appropriate.

Reply:
replaced with `<Loading />` in 4944f3417cc3a961b2743f9c4db7316d816602dc